floodlight/io/dfl.py

```diff
@@ -0,0 +1,324 @@
+from pathlib import Path
+from typing import Dict, Tuple, Union
+
+from lxml import etree
+import datetime as dt
+import numpy as np
+
+from floodlight.core.code import Code
+from floodlight.core.pitch import Pitch
+from floodlight.core.xy import XY
+
+
+def _read_pitch_from_mat_info(filepath_mat_info: Union[str, Path]) -> Pitch:
+    """Reads match_information XML file and returns the playing Pitch.
+
+    Parameters
+    ----------
+    filepath_mat_info: str or pathlib.Path
+        Full path to XML File where the Match Information data in DFL format is saved.
+
+    Returns
+    -------
+    pitch: Pitch
+        Pitch object with actual pitch length and width.
+    """
+    # set up XML tree
+    tree = etree.parse(str(filepath_mat_info))
+    root = tree.getroot()
+
+    # parse pitch
+    length = root.find("MatchInformation").find("Environment").get("PitchX")
+    length = float(length) if length else None
+    width = root.find("MatchInformation").find("Environment").get("PitchY")
+    width = float(width) if width else None
+    pitch = Pitch.from_template(
+        "dfl",
+        length=length,
+        width=width,
+    )
+
+    return pitch
+
+
+def _create_periods_from_dat(
+    filepath_dat: Union[str, Path]
+) -> Tuple[Dict[str, Tuple[int, int]], int]:
+    """Parses over position file and returns dictionary with periods
+
+    Parameters
+    ----------
+    filepath_dat: str or pathlib.Path
+        Path to XML File where the Position data in DFL format is saved
+
+    Returns
+    -------
+    periods: Dict[str, Tuple[int, int]
+        Dictionary with times for the segment:
+        `periods[segment] = (starttime, endtime)`
+    framerate: int
+        Temporal resolution of data in frames per second/Hertz.
+    """
+    periods = {}
+    framerate = None
+
+    # retrieve information from ball frame sets
+    for _, frame_set in etree.iterparse(filepath_dat, tag="FrameSet"):
+        if frame_set.get("TeamId") == "Ball":
+            frames = [frame for frame in frame_set.iterfind("Frame")]
+            periods[frame_set.get("GameSection")] = (
+                int(frames[0].get("N")),
+                int(frames[-1].get("N")),
+            )
+            if framerate is None:
+                delta = dt.datetime.fromisoformat(
```

Comment:
I believe the iso8601 package is a bit more robust to parse these strings

Reply:
Added as a dependency.